src/linux.rs

```
@@ -136,6 +140,30 @@ impl Command {
         self
     }
 
+    /// Reassociate child process with a namespace specified by a file
+    /// descriptor
+    ///
+    /// `file` argument is an open file referring to a namespace
+    ///
+    /// 'ns' is a namespace type
+    ///
+    /// See `man 2 setns` for further details
+    ///
+    /// Note: using `unshare` and `setns` for the same namespace is meaningless.
+    pub fn set_namespace<F: AsRawFd>(&mut self, file: &F, ns: Namespace)
+        -> io::Result<&mut Command>
+    {
+        let fd = match dup(file.as_raw_fd()) {
+            Ok(fd) => fd,
+            Err(nix::Error::Sys(errno)) => {
+                return Err(io::Error::from_raw_os_error(errno as i32));
+            },
+            Err(nix::Error::InvalidPath) => unreachable!(),
+        };
+        self.config.setns_namespaces.insert(ns, fd);
```

Here you silently replace old file descriptor without closing.